src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,273 @@
+ *
+ *     - To ensure the robot is capable of deccelerating to react to any sudden changes in
+ * the path, the trajectory is checked for backwards continuity
+ *
+ * @pre path_parameters.num_segments > 2
+ * @pre path_parameters.max_allowable_acceleration > 0
+ * @pre path_parameters.max_allowable_speed > 0
+ * @pre path_parameters.init_speed >= 0
+ * @pre path_parameters.final_speed >= 0
+ * @pre The trajectory.trajectory_elements[] array is pre-allocated to handle up to the
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS limit
+ *
+ * @param path_parameters [in] The data structure including important path parameters as
+ * defined by the FirmwareRobotPathParameters struct
+ * @param trajectory [out] The planned trajectory that follows robot dynamics limitations
+ * with the appropriate guarantees and assumptions outlines above
+ *
+ * @return The outcome of the trajectory generation. Returns OK if trajectory is valid.
+ */
+TrajectoryPlannerGenerationStatus
+app_trajectory_planner_generateConstantArcLengthSegmentation(
+    FirmwareRobotPathParameters_t path_parameters,
+    Trajectory_t trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS]);
+
+/**
+ * Returns a constant interpolation period (time) trajectory based on an input trajectory
+ *
+ * This function is intended to take a variable time trajectory (which is easier to
+ * calculate) then interpolate a constant period trajectory. This is valuable for discrete
+ * time motion controllers that operate only at constant delta-time intervals
+ *
+ * NOTE: This function does not modify the real trajectory. It will output a physically
+ * invalid trajectory if the input trajectory is invalid
+ *
+ * @pre constant_period_trajectory is pre-allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param constant_period_trajectory [out] The array to be filled with the constant time
+ * interpolation period equivalent of the input trajectory
+ * @param variable_time_trajectory [in] The valid trajectory used as
+ * reference for a constant interpolation period trajectory
+ * @param interpolation_period [in] The constant change in time that corresponds to each
+ * trajectory segment
+ *
+ */
+void app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation(
+    Trajectory_t constant_period_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    Trajectory_t variable_time_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    const float interpolation_period);
+
+
+/**
+ * Returns a the total length of any arc specified by an arc parameterizaton
+ *
+ * @param arc_length_param [in] Arc length parameterization
+ *
+ * @return [out] The arc length in meters
+ */
+static float app_trajectory_planner_getTotalArcLength(
+    ArcLengthParametrization_t arc_length_parameterization);
+
+/**
+ * Generates the X/Y points for each position on the constant arc length trajectory
+ *
+ * @pre traj_elements is pre-allocated up the the limit specified by
```

Comment:
`traj_elements` is not listed as an output in the javadoc

Reply:
Added a [in/out]